### ISSUE TITLE
'Splide' cannot be used as a JSX component. Its instance type 'Splide' is not a valid JSX element

### ISSUE THREAD

**Issue by CristolGDM** [Splidejs/react-splide#46]
### Checks

- [X] Not a duplicate.
- [X] Not a question, feature request, or anything other than a bug report directly related to React Splide. Use Discussions for these topics: https://github.com/Splidejs/splide/discussions

### Version

0.5.0 and above

### Description

Hello,

I am trying to use SplideJS on one of my projects, but somehow I am not able to use the component. I try to use it like this:

```
//...
import * as React from 'react';
import { Splide } from '@splidejs/react-splide';

export class Renderer extends React.Component<Props, RendererState> {
  //...
  renderSlideshow() {
    <Splide>
      <span>{'hello'}</span>
    </Splide>
  }
}
```

But I get the following error:

> TS2786: 'Splide' cannot be used as a JSX component. Its instance type 'Splide' is not a valid JSX element. Type 'Splide' is missing the following properties from type 'ElementClass': context, setState, forceUpdate, props, and 2 more.

\<SplideSlide> works fine, no error

This looks similar to [this issue](https://github.com/Splidejs/react-splide/issues/42), but considering the error message is different and I don't use next.js I think it's a different issue.

One thing I noticed is that the Splide type declaration is written this way:

```
import React, { ReactNode } from 'react';
export declare class Splide extends React.Component<SplideProps> {
```

Changing to this works:

```
import React, { ReactNode, Component } from 'react';
export declare class Splide extends Component<SplideProps> {
```

as does this:

```
import * as React from 'react';
export declare class Splide extends React.Component<SplideProps> {
```

I tried to roll back to previous versions, I see the issue happening every version from 0.5.0 (when that class got introduced)

My company is still using React 16 (hoping to upgrade to 17 this year 🤞 ) but I *think* there isn't breaking change on imports between 16 and 17+

I have absolutely no idea *why* Component works and not React.Component though, so if you have any suggestion on how to fix things on my side I am very open :)

### Steps to Reproduce

1. Have a dayjob where you have to replace swiper by a carousel properly supporting react
2. install any version of @splidejs/react-splide after v0.5.0
3. try to use \<Splide>

### Expected Behaviour

\<Splide> should be recognized as a proper JSX element

**Comment by NaotoshiFujita** [comment]
It usually happens because of the type definition mismatching.
https://stackoverflow.com/questions/71791347/npm-package-cannot-be-used-as-a-jsx-component-type-errors

I developed the component in react 17, and supported 18 recently by updating declarations. Use `ts-ignore` before you update the version to 17 (but 17? the latest version is 18), and then remove it after.

**Comment by amaralc** [comment]
Same issue as @CristolGDM here. In my case I didn't even upgrade react. What I did was running yarn to generate a new yarn lock file after deleting a non related dependency (pg).

yarn.lock
![image](https://user-images.githubusercontent.com/32170099/168879658-1d632b1d-a71f-438c-9538-19fa545c167a.png)

For a reason I don't understand, those changes messed up with the Splide types.

**Comment by amaralc** [comment]
Updating `@types/react` to match `react` and `react-dom` solved the issue for me.

package.json
```json
  "dependencies": {
    ...
    "react": "^18.1.0",
    "react-dom": "^18.1.0",
    ...
    },
  "devDependencies": {
    ...
     "@types/react": "^18.0.9",
    ...
  }

**Comment by NaotoshiFujita** [comment]
Yes, your solution is same with the stackoverflow's answer above.

**Comment by ranagulzar** [comment]
Hi Teams, Thanks for the help. I have fixed this problem by updating react/types package to latest.

**Comment by harikrishkk** [comment]
I recently encountered the same issue. 
`
'Splide' cannot be used as a JSX component.
  Its instance type 'Splide' is not a valid JSX element.
    Type 'Splide' is missing the following properties from type 'ElementClass': context, setState, forceUpdate, props, and 2 more.ts(2786)
`
Just a bare minimum setup for splide.

```

import * as React from 'react';
import { Splide, SplideSlide } from "@splidejs/react-splide";

export interface RatePlanSliderProps {
  setSelectedRatePlan: (id) => void;
}

export const RatePlanSlider: React.FC<any> = ({}) => {
  return (
    <div>
      <Splide>
        <SplideSlide>
          <img src="image1.jpg" alt="Image 1" />
        </SplideSlide>
        <SplideSlide>
          <img src="image2.jpg" alt="Image 2" />
        </SplideSlide>
      </Splide>
    </div>
  );
};


```
I tried setting up types as mentioned in above posts. Didnt work.  Here is my package json

```
{
    "name": "",
    "version": "1.0.0",
    "main": "",
    "types": "./dist/@types/index.d.ts",
    "private": false,
    "scripts": {
        "dev": "webpack --watch",
        "build": "webpack",
        "build:prod": "webpack --env -p",
        "build:dev": "webpack --env -d",
        "lint": "tslint -p src/tsconfig.json -c tslint.json -t stylish && tsc -p src --pretty --noEmit",
        "fix": "tslint -p src/tsconfig.json --fix -c tslint.json -t stylish",
        "bundle": "cd.. && cd my-components "
    },
    "keywords": [],
    "author": "",
    "license": "MIT",
    "husky": {
        "hooks": {
            "pre-commit": "npm run lint"
        }
    },
    "dependencies": {
        "@splidejs/react-splide": "^0.7.12",
        "@types/react-router-dom": "*",
        "@types/react": "*",
        "react-bootstrap": "^2.7.2",
        "react-dom": "^17.0.2",
        "react-hook-form": "^7.17.4"
    },
    "devDependencies": {
        "css-loader": "^6.2.0",
        "eslint-auto-fix": "^1.0.3",
        "eslint-config-prettier": "8.3.0",
        "husky": "^4.3.0",
        "node-sass": "^4.14.1",
        "prettier": "2.3.2",
        "sass-loader": "^12.1.0",
        "webpack-cli": "^4.7.2"
    },
    "peerDependencies": {
        "react": "*",
        "react-dom": "*",
        "react-intl": "*",
        "react-redux": "*",
        "redux-actions": "*",
        "redux-observable": "*",
        "react-router-dom": "*",
        "redux": "*",
        "rxjs": "*",
        "@apollo/client": "*",
        "graphql": "*"
    }
}
```
Any pointers will be of great help!
